pkg/steps/artifacts_test.go

```diff
@@ -252,13 +250,188 @@ func TestTestCaseNotifier_SubTests(t *testing.T) {
+									FinishedAt: meta.Time{Time: time.Unix(1150, 0)},
+								},
+							},
+						},
+						{
+							Name: "test",
+							State: coreapi.ContainerState{
+								Terminated: &coreapi.ContainerStateTerminated{
+									ExitCode:   1,
+									Message:    "exit message",
+									StartedAt:  meta.Time{Time: time.Unix(1200, 0)},
+									FinishedAt: meta.Time{Time: time.Unix(1250, 0)},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantTests: []*junit.TestCase{
+				{Name: "container other", Duration: 100},
+				{Name: "container test", FailureOutput: &junit.FailureOutput{Output: "exit message"}, Duration: 50},
+			},
+		},
+		{
+			name: "handles fully overlapping times",
+			pod: &coreapi.Pod{
+				ObjectMeta: meta.ObjectMeta{Annotations: map[string]string{annotationContainersForSubTestResults: "other,test"}},
+				Status: coreapi.PodStatus{
+					ContainerStatuses: []coreapi.ContainerStatus{
+						{
+							Name: "other",
+							State: coreapi.ContainerState{
+								Terminated: &coreapi.ContainerStateTerminated{
+									ExitCode:   0,
+									Message:    "success",
+									StartedAt:  meta.Time{Time: time.Unix(1050, 0)},
+									FinishedAt: meta.Time{Time: time.Unix(1150, 0)},
+								},
+							},
+						},
+						{
+							Name: "test",
+							State: coreapi.ContainerState{
+								Terminated: &coreapi.ContainerStateTerminated{
+									ExitCode:   1,
+									Message:    "exit message",
+									StartedAt:  meta.Time{Time: time.Unix(1100, 0)},
+									FinishedAt: meta.Time{Time: time.Unix(1150, 0)},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantTests: []*junit.TestCase{
+				{Name: "container other", Duration: 100},
+				{Name: "container test", FailureOutput: &junit.FailureOutput{Output: "exit message"}, Duration: 0},
+			},
+		},
+		{
+			name: "handles fully overlapping identical ",
+			pod: &coreapi.Pod{
+				ObjectMeta: meta.ObjectMeta{Annotations: map[string]string{annotationContainersForSubTestResults: "other,test"}},
+				Status: coreapi.PodStatus{
+					ContainerStatuses: []coreapi.ContainerStatus{
+						{
+							Name: "other",
+							State: coreapi.ContainerState{
+								Terminated: &coreapi.ContainerStateTerminated{
+									ExitCode:   0,
+									Message:    "success",
+									StartedAt:  meta.Time{Time: time.Unix(1000, 0)},
+									FinishedAt: meta.Time{Time: time.Unix(1100, 0)},
+								},
+							},
+						},
+						{
+							Name: "test",
+							State: coreapi.ContainerState{
+								Terminated: &coreapi.ContainerStateTerminated{
+									ExitCode:   1,
+									Message:    "exit message",
+									StartedAt:  meta.Time{Time: time.Unix(1100, 0)},
+									FinishedAt: meta.Time{Time: time.Unix(1100, 0)},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantTests: []*junit.TestCase{
+				{Name: "container other", Duration: 100},
+				{Name: "container test", FailureOutput: &junit.FailureOutput{Output: "exit message"}, Duration: 0},
```

Comment:
Why is this desired

Reply:
Having multiple overlapping containers running at the same time means you want "linear" flow.  If we suddenly flip to "parallel" flow just because of a short test, that means you can't predict what you would get.  So we are effectively treating > 1 containers as always linear flow.

Reply:
Basically:

```
When testing pods in succession it's much more useful to show the
unique time per container (setup time, then test time, then teardown
time) than to show their full duration. In the future we could add
an annotation to tune this behavior, but for now sorting and then
showing non-overlapping time is more useful.
```

We only have three cases - single container, two containers where only one is identified in the annotation, and multiple containers in the annotation.  The behavior single and two want is the same as linear time with a single container.